test/test_videoapi.py

```diff
@@ -57,34 +57,66 @@ def test_frame_reading(self):
             full_path = os.path.join(VIDEO_DIR, test_video)
 
             av_reader = av.open(full_path)
+            is_video = True if av_reader.streams.video else False
 
-            if av_reader.streams.video:
-                video_reader = VideoReader(full_path, "video")
+            if is_video:
+                av_frames, vr_frames = [], []
+                av_pts, vr_pts = [], []
+                # get av frames
                 for av_frame in av_reader.decode(av_reader.streams.video[0]):
-                    vr_frame = next(video_reader)
-
-                    assert float(av_frame.pts * av_frame.time_base) == approx(vr_frame["pts"], abs=0.1)
+                    av_frames.append(torch.tensor(av_frame.to_rgb().to_ndarray()).permute(2, 0, 1))
+                    av_pts.append(av_frame.pts * av_frame.time_base)
+                av_reader.close()
```

Comment:
Minor comment: If you use `with`, you won't need to worry about this.

Reply:
fair, addressed